include/nbl/core/math/glslFunctions.h

```diff
@@ -317,21 +315,25 @@ NBL_FORCE_INLINE T lerp(const T& a, const T& b, const U& t)
 	return core::mix<T,U>(a,b,t);
 }
 
-
 // TODO : step,smoothstep,isnan,isinf,floatBitsToInt,floatBitsToUint,intBitsToFloat,uintBitsToFloat,frexp,ldexp
 // extra note, GCC breaks isfinite, isinf, isnan, isnormal, signbit in -ffast-math so need to implement ourselves
 // TODO : packUnorm2x16, packSnorm2x16, packUnorm4x8, packSnorm4x8, unpackUnorm2x16, unpackSnorm2x16, unpackUnorm4x8, unpackSnorm4x8, packHalf2x16, unpackHalf2x16, packDouble2x32, unpackDouble2x32
 // MOVE : faceforward, reflect, refract, any, all, not
 template<typename T>
-NBL_FORCE_INLINE T dot(const T& a, const T& b);
+NBL_FORCE_INLINE T dot(const T& a, const T& b)
+{
+	static_assert(!(std::is_same_v<T, hlsl::float32_t2> || std::is_same_v<T, hlsl::float32_t3> || std::is_same_v<T, hlsl::float32_t4>));
```

Comment:
move to [`requires` clause](<https://en.cppreference.com/w/cpp/language/requires>)